pkg/pillar/cmd/loguploader/loguploader.go

```diff
@@ -0,0 +1,744 @@
+		if err != nil {
+			if unavailable {
+				contSentFailure++
+				contSentSuccess = 0
+			}
+			// if resp code is 503, or continuously 3 times unavailable failed, start to set the 'FailedToSend' status
+			// 'newlogd' gzip directory space management and random spaced out uploading schedule is
+			// based on the 'FailedToSend' status
+			if (contSentFailure >= 3) && !ctx.metrics.FailedToSend {
+				ctx.metrics.FailSentStartTime = time.Now()
+				ctx.metrics.FailedToSend = true
+				log.Functionf("doFetchSend: fail. set fail to send time %v", ctx.metrics.FailSentStartTime.String())
+				ctx.metricsPub.Publish("global", ctx.metrics)
+			}
+			log.Errorf("doFetchSend: %v got error sending http: %v", ctx.metrics.FailSentStartTime.String(), err)
+		} else {
+			if err := os.Remove(gziplogfile); err != nil {
+				log.Fatal("doFetchSend: can not remove gziplogfile", err)
+			}
+
+			contSentSuccess++
+			contSentFailure = 0
+			if contSentSuccess >= 3 && ctx.metrics.FailedToSend {
+				log.Functionf("doFetchSend: Reset failedToSend, at %v, gzip file %s is sent out ok",
+					time.Now().String(), gotFileName)
+				ctx.metrics.FailedToSend = false
+				ctx.metricsPub.Publish("global", ctx.metrics)
+			}
+			log.Tracef("doFetchSend: gzip file %s is sent out ok", gotFileName)
+		}
+		*iter++
+		return numGzipfiles - 1
+	}
+	log.Tracef("doFetchSend: does not find gz log file")
+	return 0
+}
+
+func getTimeNumber(isApp bool, fName string) (bool, int) {
+	if isApp {
+		if strings.HasPrefix(fName, types.AppPrefix) && strings.HasSuffix(fName, ".gz") {
+			fStr1 := strings.TrimPrefix(fName, types.AppPrefix)
+			fStr := strings.Split(fStr1, types.AppSuffix)
+			if len(fStr) != 2 {
+				err := fmt.Errorf("app split is not 2")
+				log.Fatal(err)
+			}
+			fStr2 := strings.TrimSuffix(fStr[1], ".gz")
+			fTime, err := strconv.Atoi(fStr2)
+			if err != nil {
+				log.Fatal(err)
+			}
+			return true, fTime
+		}
+	} else {
+		if strings.HasPrefix(fName, types.DevPrefix) && strings.HasSuffix(fName, ".gz") {
+			fStr1 := strings.TrimPrefix(fName, types.DevPrefix)
+			fStr2 := strings.TrimSuffix(fStr1, ".gz")
+			fTime, err := strconv.Atoi(fStr2)
+			if err != nil {
+				log.Fatal(err)
+			}
+			return true, fTime
+		}
+	}
+	return false, 0
+}
+
+func sendToCloud(ctx *loguploaderContext, data []byte, iter int, fName string, fTime int, isApp bool) (bool, error) {
+	size := int64(len(data))
+	log.Tracef("sendToCloud: size %d, isApp %v, iter %d", size, isApp, iter)
+
+	buf := bytes.NewBuffer(data)
+	if buf == nil {
+		log.Fatal("sendToCloud malloc error:")
+	}
+
+	var logsURL string
+	var sentFailed, serviceUnavailable bool
+	if isApp {
+		fStr1 := strings.TrimPrefix(fName, types.AppPrefix)
+		fStr := strings.Split(fStr1, types.AppSuffix)
+		if len(fStr) != 2 {
+			err := fmt.Errorf("app split is not 2")
+			log.Fatal(err)
+		}
+		appUUID := fStr[0]
+		appLogURL := fmt.Sprintf("apps/instanceid/%s/newlogs", appUUID)
```

Comment:
Do we need the same logic as below? https://github.com/lf-edge/eve/blob/71d286e6fedbf41b795fbc463473ccb7d14493a6/pkg/pillar/cmd/logmanager/logmanager.go#L1029-L1036

Reply:
@giggsoff if you were referring to the v1 vs v2 logic, this newlog scheme only supports V2API

Reply:
Please take a look https://github.com/lf-edge/adam/pull/50#discussion_r532398858